scalyr_agent/copying_manager/checkpoints.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2014-2020 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+import os
+
+if False:
+    from typing import Dict
+    from typing import Optional
+
+from scalyr_agent import util as scalyr_util
+from scalyr_agent.scalyr_logging import AgentLogger
+from scalyr_agent.configuration import Configuration
+
+import six
+
+"""
+The main idea of the checkpoint files is to store the states of the currently running LogFileProcessors in files, so
+the agent can continue reading files from the point where it has stopped.
+
+Since the log copying activity is distributed among worker sessions, those sessions have their own checkpoint files.
+The overall state of the particular worker session is periodically written to the checkpoint file
+named "checkpoints-worker-<WORKER_SESSION_ID>.json". This file contains the checkpoint states for every log file processor
+in the worker session.
+
+Besides that file, there is also the "active-checkpoints" file. This file contains only checkpoint states for the log processors
+that have been active during the current worker session iteration. This is needed to save the latest progress as soon as
+possible to minimize the chance of data loss if something unexpected happens.
+
+Before the agent starts, it looks for all checkpoint files that may be left from the previous run, reads and writes them
```

Comment:
I would rephrase this sentence as "When the agent starts but before the individual workers start executing, the agent looks for all checkpoint files..."